pkg/gardenlet/bootstrap/util/util.go

```diff
@@ -102,7 +105,18 @@ func UpdateGardenKubeconfigCAIfChanged(ctx context.Context, log logr.Logger, see
 		return nil, fmt.Errorf("invalid kubeconfig: currently set authinfo %s not found among authinfos", curContext.AuthInfo)
 	}
 
-	if bytes.Equal(curCluster.CertificateAuthorityData, gardenClientConnection.GardenClusterCACert) {
+	gardenClusterCACert := gardenClientConnection.GardenClusterCACert
+	if len(gardenClusterCACert) == 0 && gardenAPIReader != nil {
+		log.Info("Getting CA from the garden cluster")
+		var kubeRootCA corev1.ConfigMap
+		if err := gardenAPIReader.Get(ctx, client.ObjectKey{Namespace: core.GardenerSystemPublicNamespace, Name: kubeRootCAConfigMapName}, &kubeRootCA); err != nil {
```

Comment:
```suggestion
		kubeRootCAConfigMap := &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Namespace: core.GardenerSystemPublicNamespace, Name: kubeRootCAConfigMapName}}
		if err := gardenAPIReader.Get(ctx, client.ObjectKeyFromObject(kubeRootCAConfigMap), kubeRootCAConfigMap); err != nil {
```

Now you can remove the `kubeRootCAConfigMapName` const (the name is only used here)